ixmp4/data/db/run/filter.py

```diff
@@ -13,15 +13,27 @@ class IamcRunFilter(filters.BaseFilter, metaclass=filters.FilterMeta):
     def join(self, exc, session=None):
         if not utils.is_joined(exc, TimeSeries):
             exc = exc.join(TimeSeries, onclause=TimeSeries.run__id == Run.id)
-
-        model = get_datapoint_model(session)
-        if not utils.is_joined(exc, model):
-            exc = exc.join(model, onclause=model.time_series__id == TimeSeries.id)
         return exc
 
 
 class RunFilter(base.RunFilter, metaclass=filters.FilterMeta):
-    iamc: IamcRunFilter | filters.Boolean
+    iamc: IamcRunFilter | filters.Boolean | None = None
+
+    def join_datapoints(self, exc: db.sql.Select, session=None):
+        if not utils.is_joined(exc, TimeSeries):
+            exc = exc.join(TimeSeries, onclause=TimeSeries.run__id == Run.id)
+        return exc
+
+    def filter_iamc(self, exc, c, v, session=None):
```

Comment:
I don't see `c` being used anywhere in the function, is that correct? 
Why don't we typehint `exc, c, v` to make more explicit what we expect here?

Reply:
Happy to review or write a cleanup PR.
These functions are never type hinted, but i would adjust them /all/ while we are at it, not just this one.